apps/discord-bot/src/listeners/events/auto-thread.test.ts

```diff
@@ -0,0 +1,145 @@
+import { Client, Events, TextChannel } from "discord.js";
+import {
+  mockTextChannel,
+  mockMessage,
+  mockGuildMember,
+  emitEvent,
+} from "@answeroverflow/discordjs-mock";
+import { setupAnswerOverflowBot } from "~discord-bot/test/sapphire-mock";
+import { randomSnowflake } from "@answeroverflow/discordjs-utils";
+import { mockPublicThread } from "@answeroverflow/discordjs-mock";
+import { createChannel, createServer } from "@answeroverflow/db";
+import { toAOServer, toAOChannel } from "~discord-bot/utils/conversions";
+
+let client: Client;
+let textChannel: TextChannel;
+
+beforeEach(async () => {
+  client = await setupAnswerOverflowBot();
+  textChannel = mockTextChannel(client);
+});
+
+const mockMessageCorrect = async () => {
+  // Channel with flag enabled
+  await createServer(toAOServer(textChannel.guild));
+  await createChannel({
+    ...toAOChannel(textChannel),
+    flags: { sendMarkSolutionInstructionsInNewThreads: true },
```

Comment:
lets swap this over to auto thread now

Reply:
Can we do this in the actual code as well?

Reply:
yeah we can, i'll be merging the new channel menu soon so thatll let us enable it

Reply:
Good good 👍